src/main/host/descriptor/socket/unix.rs

```diff
@@ -1320,6 +1444,17 @@ impl UnixSocketCommon {
     }
 }
 
+fn backlog_to_queue_size(backlog: i32) -> u32 {
+    // linux also makes this cast, so negative backlogs wrap around to large positive backlogs
+    // https://elixir.free-electrons.com/linux/v5.11.22/source/net/unix/af_unix.c#L628
+    let backlog = backlog as u32;
+
+    // the linux '__sys_listen()' applies the somaxconn max to all protocols, including unix sockets
+    let queue_limit = std::cmp::min(backlog, unsafe { c::SHADOW_SOMAXCONN }.try_into().unwrap());
+
+    queue_limit.saturating_add(1)
```

Comment:
Why +1? (Probably worth a comment)

Reply:
Just because that's what Linux does :) But I'll add a comment here.